addons/test_mail/tests/test_mail_followers.py

```diff
@@ -417,62 +414,154 @@ def test_auto_subscribe_notify_email(self):
-        common_partner = self.env['res.partner'].create({"name": "demo1", "email": "demo1@test.com"})
-        user_1 = self.env['res.users'].create({'login': 'demo1', 'partner_id': common_partner.id, 'notification_type': 'email'})
-        user_2 = self.env['res.users'].create({'login': 'demo2', 'partner_id': common_partner.id, 'notification_type': 'inbox'})
+    @classmethod
+    def setUpClass(cls):
+        super(RecipientsNotificationTest, cls).setUpClass()
+
+        # portal user for testing share status / internal subtypes
+        cls.user_portal = cls._create_portal_user()
+        cls.partner_portal = cls.user_portal.partner_id
+
+        # simple customer
+        cls.customer = cls.env['res.partner'].create({
+            'email': 'customer@test.customer.com',
+            'name': 'Customer',
+            'phone': '+32455778899',
+        })
 
-        #Trigger auto subscribe notification
-        test = self.env['mail.test.track'].create({"name": "Test Track", "user_id": user_2.id})
+        # Simulate case of 2 users that got their partner merged
+        cls.common_partner = cls.env['res.partner'].create({
+            'email': 'common.partner@test.customer.com',
+            'name': 'Common Partner',
+            'phone': '+32455998877',
+        })
+        cls.user_1, cls.user_2 = cls.env['res.users'].with_context(no_reset_password=True).create([
+            {'groups_id': [(4, cls.env.ref('base.group_portal').id)],
+             'login': '_login_portal',
+             'notification_type': 'email',
+             'partner_id': cls.common_partner.id,
+            },
+            {'groups_id': [(4, cls.env.ref('base.group_user').id)],
+             'login': '_login_internal',
+             'notification_type': 'inbox',
+             'partner_id': cls.common_partner.id,
+            }
+        ])
+
+    def assertRecipientsData(self, recipients_data, records, partners):
+        """ Custom assert as recipients structure is custom and may change due
+        to some implementation choice. Currently supporting only single-record
+        computation. """
+        self.assertEqual(len(recipients_data), len(partners.ids))
+        for partner in partners:
+            partner_data = next(r for r in recipients_data if r[0] == partner.id)
+            (_pid, active, pshare, notif, groups) = partner_data
+            user = partner.user_ids[-1] if partner.user_ids else self.env['res.users']
+            self.assertEqual(active, partner.active)
+            self.assertEqual(set(groups), set(user.groups_id.ids) if user else {None})
+            self.assertEqual(notif, user.notification_type if user else None)
+            self.assertEqual(pshare, partner.partner_share)
+
+    @users('employee')
+    def test_notification_nodupe(self):
+        """ Check that we only create one mail.notification per partner. Post
+        install because we need the registry to be ready to send notification.
```

Comment:
Shouldn't this "post_install" reference be at class level since it's defined there?